processors/data_contributions/processor.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import requests
+import logging
+from .. import base
+from . import extractors as extractors_module
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    def convert_ids(contrib):
+        converted_ids = {key: val.hex for key, val in contrib.items()
+                        if val.__class__.__name__ == 'UUID'}
+        contrib.update(converted_ids)
+        return contrib
+
+    extractors = base.helpers.get_variables(
+        extractors_module, lambda x: x.startswith('extract_'))
+
+    # Upsert source
+    source = extractors['extract_source']()
+    source_id = base.writers.write_source(conn, source)
+
+    # Upsert documents from data contributions
+    for_processing = """SELECT * FROM data_contributions
+                        WHERE approved=true
+                        AND trial_id IS NOT NULL
+                        AND document_category_id IS NOT NULL
+                        AND (data_url IS NOT NULL OR url IS NOT NULL);"""
+    for data_contrib in conn['explorer'].query(for_processing):
+        data_contrib = convert_ids(data_contrib)
+        trial = conn['database']['trials'].find_one(id=data_contrib['trial_id'])
+        if not trial:
+            continue
+        _process_document(conn, data_contrib, extractors, source_id)
+
+    # Remove documents from data_contributions
+    for_removal = """SELECT * FROM data_contributions
+                     WHERE approved=false
+                     AND document_id IS NOT NULL;"""
```

Comment:
It's harmless and no problem in leaving it in, but the trailing semicolon is unnecessary here.

Please use single quotes and format the queries in separate lines, as:
```python
for_removal = '''
    SELECT * FROM data_contributions
    WHERE approved = false
    AND document_id IS NOT NULL
'''
```